packages/grid/_modules_/grid/hooks/features/pagination/useGridPage.ts

```diff
@@ -151,9 +159,25 @@ export const useGridPage = (
   /**
    * EFFECTS
    */
+  React.useEffect(() => {
+    if (process.env.NODE_ENV !== 'production') {
+      if (props.paginationMode === 'server' && props.rowCount == null) {
+        noRowCountInServerMode();
```

Comment:
Should we only warn once? We seem to be doing it in the other places.

Reply:
The check to only warn once is hidden inside `buildWarning`